pytlu/tlu_eudaq.py

```diff
@@ -0,0 +1,332 @@
+                time.sleep(additional_delay)
+            last_readout_time = time.time()
+            last_timestamp_start = t_start
+
+            actual_data = raw_data[i_start:i_stop]
+
+            skipped_triggers = meta_data[i]['skipped_triggers']
+            for dat in actual_data:
+                    actual_trigger_number = dat['trigger_id']
+                    trg_number, trg_timestamp = dat['trigger_id'], dat['time_stamp']
+                    # Check for jumps in trigger number
+                    if actual_trigger_number != last_trigger_number + 1:
+                        logging.warning('Expected != Measured trigger number: %d != %d', last_trigger_number + 1, actual_trigger_number)
+                    last_trigger_number = actual_trigger_number
+
+                    yield trg_number, trg_timestamp, skipped_triggers
+
+
+def main():
+    chip = None
+
+    args = tlu.parse_arguments(eudaq=True)
+
+    # Create configuration dict
+    config = tlu.create_configuration(args)
+
+    # Import EUDAQ python wrapper with error handling
+    try:
+        from PyEUDAQWrapper import PyTluProducer
+    except ImportError:
+        if not config['path']:
+            logging.error('Cannot find PyEUDAQWrapper! Please specify the path of your EUDAQ installation!')
+            return
+        else:
+            wrapper_path = os.path.join(config['path'], 'python/')
+            sys.path.append(os.path.join(config['path'], 'python/'))
+            try:
+                from PyEUDAQWrapper import PyTluProducer
+            except ImportError:
+                logging.error('Cannot find PyEUDAQWrapper in %s', wrapper_path)
+                return
+
+    logging.info('Connect to %s', config['address'])
+
+    if config['replay']:
+        if os.path.isfile(config['replay']):
+            logging.info('Replay %s', config['replay'])
+        else:
+            logging.error('Cannot open %s for replay!', config['replay'])
+    delay = config['delay'] if config['delay'] else 0.
+
+    pp = PyTluProducer(config['address'])
+
+    def get_tx_state():
+        if chip is not None:
+            tx_state = chip['tlu_master'].TX_STATE
+        else:
+            return
+        tx_state_str = []
+        for i in range(6):
+            if in_en & (0x20 >> i):
+                tx_state_str.append(" %x" % ((tx_state >> (4 * i)) & 0xF))
+            else:
+                tx_state_str.append(" -")
+        return ",".join(tx_state_str)
+
+    def send_data_to_eudaq(data, skipped_triggers, event_counter):
+        '''
+        Send data to EUDAQ.
+
+        Parameters:
+        ------------
+            data: tuple
+                Tuple containing (trigger number, trigger timestamp, skipped triggers) of actual readout data.
+            event_counter: int
+                Event number (number of received triggers)
+        '''
+        trg_number, trg_timestamp = data['trigger_id'], data['time_stamp']
```

Comment:
EUDAQ expects 15 bit trigger number and we send 31-bit? Maybe an issue?

Reply:
Is it 15 bit or 16 bit? By default (see `--n_bits_trig_id` option) trigger number is 16 bit.

Reply:
Depends on the firmware. But doesn't matter if you clock 32 times. With that, you are on the safe side. The word is still correct (whether it is 15 or 16 or 32 bit). This is because the LSB comes first.

Disadvantage is that it is generally slower.

Reply:
It depends on the FW of the device which has to clock out the triger number. But pytlu can produce any trigger number length (smller than 31 bit), it can be specified with the `--n_bits_trig_id` option.

Reply:
I mean, it could be an issue to send a number larger than 16-bit to EUDAQ.

Reply:
Did you try this with EUDAQ? Sending a trigger number > 16 bit? Or did you check code that it is 32-bit? I tried finding the position where this is done, but could not quickly spot it

Reply:
I asked Andre Rummler if it would be okay to send a trigger number larger than 16 bit, lets see what he will tell us. In case EUDAQ cannot handle trigger numbers larger than 16 bit I would just fix it to 16 bit for the pytlu producer.